js/src/behaviors/dismiss.ts

```diff
@@ -0,0 +1,39 @@
+export function initDismiss() {
+	window.addEventListener('click', ({ target }: Event) => {
+		const dismissButton = target instanceof Element && target.closest('[data-dismiss]');
+
+		if (!dismissButton) {
+			return;
+		}
+
+		const dismissTarget = dismissButton.closest('[data-dismissable]');
+
+		if (!dismissTarget) {
+			return;
+		}
+
+		dismissElement(dismissTarget);
+	});
+}
+
+async function dismissElement(dismissTarget: Element) {
+	dismissTarget.addEventListener('animationend', () => {
+		dismissTarget.remove();
+	});
+
+	const dismissAnimation = dismissTarget.getAttribute('data-dismiss-animation');
+
+	if (dismissAnimation) {
+		switch (dismissAnimation) {
+			case 'slide-up':
+				dismissTarget.classList.add('animation-slide-up');
+				break;
+			case 'fade':
+				dismissTarget.classList.add('animation-fade');
+				break;
+		}
+	}
```

Comment:
I noticed a case I missed with my suggestion: when there's no specified animation. That's on me, sorry!

I think it might be worth an `else` check here.

```suggestion
	} else {
		dismissTarget.remove();
    }
```